src/capture/mod.rs

```diff
@@ -90,10 +91,29 @@ impl State for Dead {}
 /// ```
 pub struct Capture<T: State + ?Sized> {
     nonblock: bool,
-    handle: NonNull<raw::pcap_t>,
+    handle: Arc<Handle>,
     _marker: PhantomData<T>,
 }
 
+struct Handle {
+    handle: NonNull<raw::pcap_t>,
+}
+
+impl Handle {
+    fn as_ptr(&self) -> *mut raw::pcap_t {
+        self.handle.as_ptr()
+    }
+}
+
+unsafe impl Send for Handle {}
+unsafe impl Sync for Handle {}
+
+impl Drop for Handle {
+    fn drop(&mut self) {
+        unsafe { raw::pcap_close(self.handle.as_ptr()) }
+    }
+}
+
 // A Capture is safe to Send as it encapsulates the entire lifetime of `raw::pcap_t *`, but it is
 // not safe to Sync as libpcap does not promise thread-safe access to the same `raw::pcap_t *` from
 // multiple threads.
```

Comment:
If the Handle implements `Send` then the explicit `impl Send` for `Capture` might no longer be necessary. In that case move the comment that explains the Send/Sync decision to be just above the Send impl for the Handle.